### PR TITLE
feat(vault): Added vault configuration endpoint DELETE

### PR DESCRIPTION
**Pull request by dmyios**
## Description

Added vault configuration endpoint DELETE

**Fixes** # MON-14984

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [x] 23.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/src/Core/Security/Vault/Application/UseCase/DeleteVaultConfiguration/DeleteVaultConfiguration.php [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+                );
+
+                return;
+            }
+
+            $this->writeVaultConfigurationRepository->delete($deleteVaultConfigurationRequest->vaultConfigurationId);
+            $presenter->setResponseStatus(new NoContentResponse());
+        } catch (\Throwable $ex) {
+            $this->error(
+                'An error occured in while updating vault configuration',
```
Comment (jeremyjaouen) [inline centreon/src/Core/Security/Vault/Application/UseCase/DeleteVaultConfiguration/DeleteVaultConfiguration.php:104]:
```suggestion
                'An error occured in while deleting vault configuration',
```

File: centreon/src/Core/Security/Vault/Application/UseCase/DeleteVaultConfiguration/DeleteVaultConfigurationRequest.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ */
+
+declare(strict_types=1);
+
+namespace Core\Security\Vault\Application\UseCase\DeleteVaultConfiguration;
+
+final class DeleteVaultConfigurationRequest
+{
+    /** @var int */
+    public int $vaultConfigurationId = 1;
```
Comment (jeremyjaouen) [inline centreon/src/Core/Security/Vault/Application/UseCase/DeleteVaultConfiguration/DeleteVaultConfigurationRequest.php:29]:
```suggestion
    public int $vaultConfigurationId = 0;
```

File: centreon/src/Core/Security/Vault/Application/UseCase/DeleteVaultConfiguration/DeleteVaultConfigurationRequest.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    /** @var int */
+    public int $typeId = 1;
```
Comment (jeremyjaouen) [inline centreon/src/Core/Security/Vault/Application/UseCase/DeleteVaultConfiguration/DeleteVaultConfigurationRequest.php:32]:
```suggestion
    public int $typeId = 0;
```

File: centreon/tests/api/features/VaultConfiguration.feature [hunk 4]
```diff
@@ -208,3 +208,54 @@ Feature: Vault Configuration API
...
+        "address": "127.0.0.1",
+        "port": 8200,
+        "storage": "myStorageFolder",
+        "role_id": "myRoleId",
+        "secret_id": "mySecretId"
+      }
+    """
+
+    When I send a DELETE request to '/api/latest/administration/vaults/1/configurations/2'
+    Then the response code should be "404"
```
Comment (jeremyjaouen) [inline centreon/tests/api/features/VaultConfiguration.feature:261]: Add a test for Delete vault configuration as an non admin user

**Review by jeremyjaouen — CHANGES_REQUESTED** [review]

File: centreon/tests/api/features/VaultConfiguration.feature (hunk 4 above)
Comment (dmyios) [reply centreon/tests/api/features/VaultConfiguration.feature:284]: Done

**Review by jeremyjaouen — APPROVED** [review]

**Review by abuathier — APPROVED** [review]

**Pull request merged**